hibernate-core/src/main/java/org/hibernate/generator/Generator.java

```diff
@@ -34,6 +34,9 @@
  *	   SQL {@code select}, though in certain cases this additional round trip may be avoided.
  *	   An important example is id generation using an identity column.
  * </ul>
+ * Note that it's also possible for a Generator to implement both interfaces and determine the timing
```

Comment:
"A Generator may implement" is enough.

"Note that it's also possible for" is not really doing any useful work here.

Reply:
Thanks for the suggestion, applied it.